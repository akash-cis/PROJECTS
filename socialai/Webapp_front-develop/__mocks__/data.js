export default [
  {
    userName: "Funnel AI Admin",
    date: "2020-01-22",
    userId: 1,
    status: "Viewed",
    count: 19,
    source: "Prospect",
  },
  {
    userName: "Pain",
    date: "2020-01-23",
    userId: 1,
    status: "Saved",
    count: 2,
    source: "Prospect",
  },
  {
    userName: "Love",
    date: "2020-01-23",
    userId: 1,
    status: "Viewed",
    count: 10,
    source: "Prospect",
  },
  {
    userName: "Beauty",
    date: "2020-01-13",
    userId: 1,
    status: "Accepted",
    count: 6,
    source: "Prospect",
  },
  {
    userName: "Virtue",
    date: "2019-12-20",
    userId: 1,
    status: "Accepted",
    count: 6,
    source: "Prospect",
  },
]
