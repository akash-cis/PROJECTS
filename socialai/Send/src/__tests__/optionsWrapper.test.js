import React from "react"
import { render, fireEvent, waitForElement } from "@testing-library/react"
import OptionsWrapper, {
  mapify,
  rangeFilter,
  filterByRange,
  substractDays,
} from "../library/optionsWrapper"
import data from "data"

test("renders dropdown with period picker and modifies state", async () => {
  const { getByTestId, getByText } = render(
    <OptionsWrapper periods>{props => <p {...props}>Chart</p>}</OptionsWrapper>
  )
  fireEvent.mouseOver(getByTestId("period"))
  await waitForElement(() => fireEvent.click(getByText("1 week")))
  expect(getByTestId("period")).toHaveTextContent("1 week")
})

test("renders dropdown with change picker and modifies state", async () => {
  const { getByTestId, getByText } = render(
    <OptionsWrapper changes>{props => <p {...props}>Chart</p>}</OptionsWrapper>
  )
  fireEvent.mouseOver(getByTestId("changes"))
  await waitForElement(() => fireEvent.click(getByText("This quarter vs last")))
  expect(getByTestId("changes")).toHaveTextContent("This quarter vs last")
})

// TOOD test range
// test("changes range according to selected period", () => {
//   const { getByTestId, getByText } = render(<OptionsWrapper></OptionsWrapper>)
//   rangeFilter(data, 11, new Date(2020, 0, 23)))
// })

test("changes period (range) according to selected period", async () => {
  const { getByTestId, getByText, debug } = render(
    <OptionsWrapper periods date="2020-01-23" data={data} lambda={data => data}>
      {({ data }) => (
        <div data-testid="data">
          {data.reduce((acc, curr) => acc + curr.count, 0)}
        </div>
      )}
    </OptionsWrapper>
  )
  // starts with range 30
  expect(getByTestId("data")).toHaveTextContent("37")
  fireEvent.mouseOver(getByTestId("period"))

  // range 7
  await waitForElement(() => fireEvent.click(getByText("1 week")))
  expect(getByTestId("data")).toHaveTextContent("31")

  // range 1
  await waitForElement(() => fireEvent.click(getByText("3 months")))
  expect(getByTestId("data")).toHaveTextContent("43")
})

// test("changes change (range) according to selected change", async () => {
//   const { getByTestId, getByText, debug } = render(
//     <OptionsWrapper change data={data} lambda={data => data}>
//       {({ data }) => (
//         <div data-testid="data">
//           {data.reduce((acc, curr) => acc + curr.count, 0)}
//         </div>
//       )}
//     </OptionsWrapper>
//   )
//   // starts with range 2
//   expect(getByTestId("data")).toHaveTextContent("1")
//   fireEvent.mouseOver(getByTestId("period"))

//   // range 7
//   await waitForElement(() => fireEvent.click(getByText("1 week")))
//   expect(getByTestId("data")).toHaveTextContent("31")

//   // range 1
//   await waitForElement(() => fireEvent.click(getByText("1 day")))
//   expect(getByTestId("data")).toHaveTextContent("12")
// })

const changeMenu = {
  TODAY_YESTERDAY: { label: "Today vs yesterday", range: 2 },
  MONTH_LAST: { label: "This month vs last", range: 60 },
  WEEK_LAST: { label: "This week vs last", range: 14 },
}

test("maps object to menu", () => {
  expect(mapify(changeMenu)).toMatchSnapshot()
})

test("shows nothing on zero days", () => {
  expect(filterByRange(data, 0, "2020-01-23T00:00:00")).toMatchSnapshot()
})

test("filters correctly by 1 week range", () => {
  expect(filterByRange(data, 7, "2020-01-23T00:00:00")).toMatchSnapshot()
})

test("filters correctly by 30 days range", () => {
  expect(filterByRange(data, 30, "2020-01-23T00:00:00")).toMatchSnapshot()
})

test("substract days to date", () => {
  substractDays("2020-01-01T00:00:00", 6)
})
