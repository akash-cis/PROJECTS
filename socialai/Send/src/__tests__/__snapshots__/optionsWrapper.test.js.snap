// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters correctly by 1 week range 1`] = `
Array [
  Object {
    "count": 19,
    "date": "2020-01-22",
    "source": "Prospect",
    "status": "Viewed",
    "userId": 1,
    "userName": "Funnel AI Admin",
  },
  Object {
    "count": 2,
    "date": "2020-01-23",
    "source": "Prospect",
    "status": "Saved",
    "userId": 1,
    "userName": "Pain",
  },
  Object {
    "count": 10,
    "date": "2020-01-23",
    "source": "Prospect",
    "status": "Viewed",
    "userId": 1,
    "userName": "Love",
  },
]
`;

exports[`filters correctly by 30 days range 1`] = `
Array [
  Object {
    "count": 19,
    "date": "2020-01-22",
    "source": "Prospect",
    "status": "Viewed",
    "userId": 1,
    "userName": "Funnel AI Admin",
  },
  Object {
    "count": 2,
    "date": "2020-01-23",
    "source": "Prospect",
    "status": "Saved",
    "userId": 1,
    "userName": "Pain",
  },
  Object {
    "count": 10,
    "date": "2020-01-23",
    "source": "Prospect",
    "status": "Viewed",
    "userId": 1,
    "userName": "Love",
  },
  Object {
    "count": 6,
    "date": "2020-01-13",
    "source": "Prospect",
    "status": "Accepted",
    "userId": 1,
    "userName": "Beauty",
  },
]
`;

exports[`maps object to menu 1`] = `
Array [
  Array [
    "TODAY_YESTERDAY",
    "Today vs yesterday",
  ],
  Array [
    "MONTH_LAST",
    "This month vs last",
  ],
  Array [
    "WEEK_LAST",
    "This week vs last",
  ],
]
`;

exports[`shows nothing on zero days 1`] = `Array []`;
